wfe/wfe.go

```diff
@@ -169,6 +169,7 @@ func (wfe *WebFrontEndImpl) HandleFunc(
 	mux *http.ServeMux,
 	pattern string,
 	handler wfeHandlerFunc,
+	includeTrailingSlash bool,
```

Comment:
I'm generally wary of adding too many parameters to functions, especially boolean parameters (because it's hard to figure out when looking at the call site what the boolean does).

I think in this case, rather than adding `includeTrailingSlash`, you can just change the `pattern` parameter. So, for instance, you would change `RootCertPath` from `/root` to `/root/`. That should allow you to revert these changes to `HandleFunc` and simplify this PR.

Reply:
The disadvantage is that current users of Pebble which rely that the root can be get via `/root` will suddenly no longer work, since `/root` will yield 404 and `/root/` must be used. (Same for the new private key / intermediate certificate endpoints.) If that's not a concern, I don't mind changing this (as it indeed would simplify everything a lot).

Reply:
Good point about the existing clients that hardcode `/root`. How about this: make the new path `/roots/0` `/roots/1`, etc, then add a redirect from `/root` to `/roots/1`.

The addition of the `s` works around the fact that `net/http`'s ServeMux automatically redirects `foo` to `foo/` if `foo/` is registered as a subtree.

Reply:
Thanks for the fixes so far! Waiting on an update to this feedback before doing my next round of review.

Reply:
Oops, I see you just pushed some more changes, including these! Ready for re-review?

Reply:
I've implemented this almost, what I've left is that's there `/roots/`, `/roots/alternate/1`, etc. I could change this, but then `handleCert`/`handleKey` would work differently than `Certificate`, in particular adding the link headers.

Reply:
Sorry, I'm still in the process of restoring my main dev machine from backups, so I'm a bit slow today :) Should be ready for re-review now!

Reply:
No problem, I think your turnaround is very fast! I just wanted to communicate clearly so we didn't both wind up waiting on each other. :-)